### PR TITLE
fix: Update runners for each workflow job

### PR DESCRIPTION
**Pull request by samuelarogbonlo**

### CODE REVIEW

File: .github/workflows/docker-hub-parachain.yml [hunk 1]
```diff
@@ -8,7 +8,7 @@ on:
 
 jobs:
   hadolint:
-    runs-on: ubuntu-20.04
+    runs-on: zeitgeist-runner
```
Comment (sea212) [inline .github/workflows/docker-hub-parachain.yml:11]:
```suggestion
    runs-on: [self-hosted, zeitgeist-runner]
```

> Although the self-hosted label is not required, we strongly recommend specifying it when using self-hosted runners to ensure that your job does not unintentionally specify any current or future GitHub-hosted runners.

https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/using-self-hosted-runners-in-a-workflow#about-self-hosted-runner-labels

File: .github/workflows/docker-hub-parachain.yml [hunk 2]
```diff
@@ -19,7 +19,7 @@ jobs:
   publish:
     needs: hadolint
     name: Publish
-    runs-on: ubuntu-latest
+    runs-on: zeitgeist-runner
```
Comment (sea212) [inline .github/workflows/docker-hub-parachain.yml:22]:
```suggestion
    runs-on: [self-hosted, zeitgeist-runner]
```

> Although the self-hosted label is not required, we strongly recommend specifying it when using self-hosted runners to ensure that your job does not unintentionally specify any current or future GitHub-hosted runners.

https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/using-self-hosted-runners-in-a-workflow#about-self-hosted-runner-labels

File: .github/workflows/docker-hub.yml [hunk 3]
```diff
@@ -8,7 +8,7 @@ on:
 
 jobs:
   hadolint:
-    runs-on: ubuntu-20.04
+    runs-on: zeitgeist-runner
```
Comment (sea212) [inline .github/workflows/docker-hub.yml:11]:
```suggestion
    runs-on: [self-hosted, zeitgeist-runner]
```

> Although the self-hosted label is not required, we strongly recommend specifying it when using self-hosted runners to ensure that your job does not unintentionally specify any current or future GitHub-hosted runners.

https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/using-self-hosted-runners-in-a-workflow#about-self-hosted-runner-labels

File: .github/workflows/docker-hub.yml [hunk 4]
```diff
@@ -19,7 +19,7 @@ jobs:
   publish:
     needs: hadolint
     name: Publish
-    runs-on: ubuntu-latest
+    runs-on: zeitgeist-runner
```
Comment (sea212) [inline .github/workflows/docker-hub.yml:22]:
```suggestion
    runs-on: [self-hosted, zeitgeist-runner]
```

> Although the self-hosted label is not required, we strongly recommend specifying it when using self-hosted runners to ensure that your job does not unintentionally specify any current or future GitHub-hosted runners.

https://docs.github.com/en/actions/hosting-your-own-runners/managing-self-hosted-runners/using-self-hosted-runners-in-a-workflow#about-self-hosted-runner-labels

**Review by sea212 — COMMENTED** [review]
The following jobs are also failing due to missing disk space and require a self hosted runner:

Workflow: Rust
Jobs:
- Quick check benchmarks
- Clippy
- Test parachain build

File: .github/workflows/rust.yml [hunk 5]
```diff
@@ -96,15 +98,17 @@ jobs:
...
+    runs-on: [self-hosted, zeitgeist-runner]
     steps:
       - name: Checkout repository
         uses: actions/checkout@v2
 
       - name: Install build tools
         run: ./scripts/init.sh
       
-      - run: ./scripts/benchmarks/quick_check.sh
+      - run: |
```
Comment (maltekliemann) [inline .github/workflows/rust.yml:109]:
```suggestion
      - name: Check benchmarks
        run:
```

**Review by maltekliemann — APPROVED** [review]

**Comment by Chralt98** [comment]
What are the CI failures here? Why did you approve @maltekliemann, if you saw the CI to fail?

**Review by Chralt98 — COMMENTED** [review]
Can the following  github actions be removed then?

```
      # No disk space: https://github.com/zeitgeistpm/zeitgeist/actions/runs/5085081984/jobs/9144298675?pr=1006
      # Workaround: https://github.com/actions/runner-images/issues/2840#issuecomment-790492173
      - name: Free up disk space on GitHub hosted runners
        run: |
          # Ensure context is GitHub hosted runner
          # https://docs.github.com/en/actions/learn-github-actions/contexts#runner-context
          if [[ "${{ runner.name }}" == "GitHub Actions"* ]]; then
            echo "Freeing up space in GitHub hosted runner"
            sudo rm -rf /usr/share/dotnet
            sudo rm -rf /opt/ghc
            sudo rm -rf "/usr/local/share/boost"
            sudo rm -rf "$AGENT_TOOLSDIRECTORY"
          fi
```

**Comment by sea212** [comment]
> What are the CI failures here? Why did you approve @maltekliemann, if you saw the CI to fail?

Previously the CI has passed. We'll have to look into it.

**Comment by sea212** [comment]
Pipeline successfully runs through again using GitHub hosted runners. Closing for now.

**Pull request closed without merging**